### PR TITLE
fixes #10122

### PR DESCRIPTION
**Pull request by Player256**
Fixes cBioPortal/cbioportal [#10122](https://github.com/cBioPortal/cbioportal/issues/10122)

Describe changes proposed in this pull request:
- added download feature in treatment per patient table
  image

Link: https://genie.cbioportal.org/study/summary?id=nsclc_public_genie_bpc

### CODE REVIEW

**Comment by inodb** [comment]
Thanks @Player256!

The downloaded data seems similar for Full and Summary Data. For e.g. this study: https://deploy-preview-4595--cbioportalfrontend.netlify.app/study/summary?id=lgg_ucsf_2014

```
Treatment	#
TMZ
10
CCNU + thalidomide
1
TMZ + accutane
1
TMZ + thalidomide
1
```

File: src/pages/studyView/tabs/SummaryTab.tsx [hunk 1]
```diff
@@ -649,6 +649,9 @@ export class StudySummaryTab extends React.Component<
                 props.filters = this.store.patientTreatmentFiltersAsStrings;
                 props.promise = this.store.patientTreatments;
                 props.onValueSelection = this.store.onTreatmentSelection;
+                props.getData = () =>
+                    this.store.getPatientTreatmentDownloadData();
+                props.downloadTypes = ['Full Data', 'Summary Data'];
```
Comment (dippindots) [inline src/pages/studyView/tabs/SummaryTab.tsx:654]: What's the difference between these two? Maybe we can just offer one available download type `Data`?

**Review by dippindots — COMMENTED** [review]
@Player256 Thank you so much for trying to fix this issue, I put some comments here, please feel free to ask me any questions regarding them.

**Review by dippindots — APPROVED** [review]
@Player256 Thanks for addressing all my comments, this looks good to me now!

**Comment by Player256** [comment]
Thank you @dippindots.

**Comment by dippindots** [comment]
@inodb It looks good to me, I will merge this one. If there is anything you want to add as an follow-up task, please feel free to  create another ticket! Thanks.

**Pull request merged**